internal/resolution/manifest/maven.go

```diff
@@ -238,6 +309,12 @@ func (m MavenManifestIO) MergeParents(ctx context.Context, result *maven.Project
 //   - identifier to locate the profile/plugin which is profile ID or plugin name
 //   - (optional) suffix indicates if this is a dependency management
 func makeRequirementVersion(dep maven.Dependency, origin string) resolve.RequirementVersion {
+	// Treat test & optional dependencies as regular dependencies to force the resolver to resolve them.
```

Comment:
I think this should be fine for now, but we may "fix" the resolver to handle `test` and `optional` dependencies if it is not that much work, and this would benefit transitive support when scanning pom.xml.